src/patterns/ec2-app.ts

```diff
@@ -180,17 +180,6 @@ export class GuEc2App {
       vpc,
       internetFacing: true,
       vpcSubnets: { subnets: GuVpc.subnetsfromParameter(scope, { type: SubnetType.PUBLIC, app }) },
-      securityGroup:
-        props.access.scope === AccessScope.RESTRICTED
-          ? new GuSecurityGroup(scope, "SecurityGroup", {
```

Comment:
Both `GuSecurityGroup` and `GuPublicInternetAccessSecurityGroup` seem to add a very permissive egress rule by default:

```
"SecurityGroupEgress": Array [
  Object {
    "CidrIp": "0.0.0.0/0",
    "Description": "Allow all outbound traffic by default",
    "IpProtocol": "-1",
  },
],
```

In the case of the EC2 App Pattern, this allows the load balancer to communicate with targets which means that everything will still work on `14.4.0`. However, this rule also allows a lot of other traffic which is unnecessary. As far as I can tell, this happens because `allowAllOutbound` [defaults to true](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-ec2.SecurityGroup.html#allowalloutbound).

I think we should consider making the following changes in separate PRs:

1. Overriding this default as part of [`GuBaseSecurityGroup`](https://github.com/guardian/cdk/blob/859ef3e6c3a0be0bab45ed4ee47ff1c684bcb278/src/constructs/ec2/security-groups/base.ts#L40-L69).
2. Removing [`GuPublicInternetAccessSecurityGroup`](https://github.com/guardian/cdk/blob/859ef3e6c3a0be0bab45ed4ee47ff1c684bcb278/src/constructs/ec2/security-groups/base.ts#L78-L87) entirely, as the AWS CDK is capable of creating a practically identical security group by default (as shown in this PR). As far as I can see this construct is currently only used by `typerighter`.